pkg/didcomm/protocol/didexchange/service_test.go

```diff
@@ -262,14 +243,23 @@ func TestService_Handle_EdgeCases(t *testing.T) {
 		require.Error(t, err)
 	})
 	t.Run("must not transition to same state", func(t *testing.T) {
-		s := &Service{outboundDispatcher: newMockOutboundDispatcher(), store: newMockStore()}
+		prov := mockProvider{}
+		ctx:= context{outboundDispatcher:prov.OutboundDispatcher(), didWallet:prov.DIDWallet()}
+		newDidDoc, err := ctx.didWallet.CreateDID()
+		require.NoError(t, err)
+
+		s := &Service{ctx:ctx, store: newMockStore()}
 		s.RegisterAutoExecute()
 		thid := randomString()
 		request, err := json.Marshal(
 			&Request{
 				Type:  ConnectionRequest,
 				ID:    thid,
 				Label: "test",
+				Connection: &Connection{
+					DID: "B.did@B:A",
```

Comment:
The value of `DID` should be `newDIDDoc.ID`

Reply:
done